alphagov/paas-cf
# [#139841751] Redirect HTTP for apps

## What

We have upgraded our HAProxy release, to enable redirecting any requests
made to http version of an app to the ssl encrypted protocol https.

We added test for absence of HSTS header when redirecting.

The RFC[1] states:

> An HSTS Host MUST NOT include the STS header field in HTTP responses
> conveyed over non-secure transport.

We therefore want to assert that one isn't set when redirecting.

And at last, we've added HTTP redirect endpoint to router ELB.

This causes all requests for the non-https version of an app to be
redirected to the corresponding https root endpoint. See [ADR444](https://github.com/alphagov/paas-team-manual#120) for details on the rationalle.

## How to review

- Code review
- Deploy `create-cloudfoundry` from this branch
- Expect all acceptance tests to pass
- Visit `healthcheck.${DEPLOY_ENV}.dev.cloudpipelineapps.digital` and witness it being redirected to `https`
- Run `curl -I healthcheck.${DEPLOY_ENV}.dev.cloudpipelineapps.digital` and expect 301

## 🚨  BEFORE MERGING :bee:

Please mind the HAProxy version is currently fixed to the dev build. This will need to be updated once the alphagov/paas-haproxy-release#9 has been merged.

## Who can review

Neither @alext nor myself.
